components/Select/Select.tsx

```diff
@@ -68,6 +72,10 @@ const StyledSelect = styled('select', {
     pointerEvents: 'none',
     color: '$selectDisabledText',
   },
+
+  '&[readonly]': {
```

Comment:
I'd recommend using [pseudo-class selector :read-only](https://developer.mozilla.org/en-US/docs/Web/CSS/:read-only)

Reply:
Readonly seems deprecated on the select element so I removed this state. https://html.com/attributes/select-readonly/